app/scripts/migrations/103.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  EthAccountType,
+  InternalAccount,
+  EthMethod,
+} from '@metamask/keyring-api';
+import { sha256FromString } from 'ethereumjs-util';
+import { v4 as uuid } from 'uuid';
+import { cloneDeep } from 'lodash';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+interface Identity {
+  name: string;
+  address: string;
+  lastSelected?: number;
+}
+
+export const version = 103;
+
+/**
+ * This migration does the following:
+ *
+ * - Creates a default state for AccountsController.
+ * - Moves identites and selectedAddress from the PreferencesController to the AccountsController state.
+ * - Removes identites and selectedAddress from the PreferencesController
```

Comment:
```suggestion
 * - Removes identities and selectedAddress from the PreferencesController
 ```

Reply:
The identities are not removed now. Updated comments in dae5a5c7050941e91b353542650e3acf1b482a5f